### PR TITLE
IGNITE-15930 Implement explicit tx argument in table API

### PR DESCRIPTION
**Pull request by sanpwc**
- Explicit tx argument added to both RecordView and KeyValueView.
- withTransaction() and transaction() are removed.

### CODE REVIEW

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java [hunk 1]
```diff
@@ -39,35 +39,38 @@
...
-     * {@link #getNullableAsync(Object)} must be used instead.
+     * {@link #getNullableAsync(Object, Transaction)} must be used instead.
      *
      * @param key A key which associated the value is to be returned. The key cannot be {@code null}.
+     * @param tx  The transaction or {@code null} to auto commit.
      * @return Future representing pending completion of the operation.
-     * @see #getNullableAsync(Object)
-     * @see #get(Object)
+     * @see #getNullableAsync(Object, Transaction)
+     * @see #getNullableAsync(Object, Transaction)
```
Comment (sk0x50) [inline modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:62]: I think it should be as follows:
```
    @see #get(Object, Transaction)
    @see #getNullableAsync(Object, Transaction)
```

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java [hunk 2]
```diff
@@ -76,38 +79,41 @@
...
      * Gets a nullable value associated with the given key.
      *
-     * <p>Note: method has same semantic as {@link #get(Object)} with regard to {@code null} values.
+     * <p>Note: method has same semantic as {@link #get(Object, Transaction)} with regard to {@code null} values.
      *
      * @param key A key which associated the value is to be returned. The key cannot be {@code null}.
+     * @param tx  The transaction or {@code null} to auto commit.
      * @return Future representing pending completion of the operation.
-     * @see #getOrDefault(Object, Object)
+     * @see #getOrDefaultAsync(Object, Object, Transaction)
```
Comment (sk0x50) [inline modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:114]: `getOrDefault(Object, Object, Transaction)`

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java [hunk 3]
```diff
@@ -291,9 +317,10 @@ default V getOrDefault(K key, V defaultValue) {
      *
      * @param key A key with which the specified value is associated. The key cannot be {@code null}.
      * @param val Value to be associated with the specified key.
+     * @param tx  The transaction or {@code null} to auto commit.
```
Comment (sk0x50) [inline modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:320]: The code snippet in the Javadoc for this method should be updated as well.
```
     * if (view.containsKey(key, tx)) {
     *   view.put(key, value, tx);
     *   return true;
     * } else {
     *   return false;
     * }</code></pre>
 ```

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java [hunk 4]
```diff
@@ -309,28 +336,31 @@ default V getOrDefault(K key, V defaultValue) {
      * @param key    A key with which the specified value is associated. The key cannot be {@code null}.
      * @param oldVal Expected value associated with the specified key.
      * @param newVal Value to be associated with the specified key.
+     * @param tx     The transaction or {@code null} to auto commit.
```
Comment (sk0x50) [inline modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:339]: please update code snippet

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java [hunk 5]
```diff
@@ -347,36 +377,40 @@ default V getOrDefault(K key, V defaultValue) {
      *
      * @param key A key with which the specified value is associated. The key cannot be {@code null}.
      * @param val Value to be associated with the specified key.
+     * @param tx  The transaction or {@code null} to auto commit.
```
Comment (sk0x50) [inline modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:380]: please update code snippet

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java (hunk 1 above)
Comment (sanpwc) [reply modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:62]: Fixed.

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java (hunk 3 above)
Comment (sanpwc) [reply modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:320]: tx param added.

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java (hunk 2 above)
Comment (sanpwc) [reply modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:114]: Fixed.

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java (hunk 4 above)
Comment (sanpwc) [reply modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:339]: tx param added.

File: modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java (hunk 5 above)
Comment (sanpwc) [reply modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java:380]: tx param added.

**Pull request closed without merging**